compoments/radioFieldSet.js

```diff
@@ -42,48 +45,55 @@ class RadioFieldSet extends Component {
       if (selectedOption.conditional) {
         if (this.conditionalValue[value]) {
           if (selectedOption.conditional.validator && !selectedOption.conditional.validator.isValid(this.conditionalValue[value])) {
-            return this.setState({error: selectedOption.conditional.validator.errorMessage})
+            return this.setState({ conditionalError: selectedOption.conditional.invalidInputErrorMessage })
           }
           return this.onSubmit({
             selected: value,
             input: this.conditionalValue[value]
           })
         }
-        return this.setState({error: 'Required'})
+        return this.setState({ conditionalError: selectedOption.conditional.emptyInputErrorMessage })
       }
       let display = selectedOption.title
-      this.onSubmit({val: this.state.value, display: display})
+      this.onSubmit({ val: this.state.value, display: display })
     } else {
-      this.setState({error: 'Required'})
+      this.setState({ error: this.errorText })
     }
   };
 
   includeOrDivider(i) {
     return this.orDivider && i == (this.options.length - 1);
   };
 
-  render(){
+  hasAnyError(){
+    return !!this.state.error || !!this.state.conditionalError;
+  }
+
+  render() {
 
     return (
       <div>
-        <div className="govuk-form-group">
+        {this.hasAnyError() &&
+          <ErrorSummary errorSummaryText={this.state.conditionalError || this.errorText} errorSummaryLocation={this.state.conditionalError ? `#${this.name}-conditional-error` : `#${this.name}-error`} />
+        }
+        <div className={this.state.error && !this.state.conditionalError ? 'govuk-form-group--error' : 'govuk-form-group'}>
```

Comment:
Is there a way of adding both of these classes when there's an error, rather than replacing. (in addition to removing the extra govuk-form-group in common below) That way you get rid of the need for this div just for spacing `<div id="before-button-content" class="govuk-!-margin-top-6"></div>`